convert2rhel/actions/__init__.py

```diff
@@ -230,63 +285,199 @@ def get_actions(actions_path, prefix):
+            skips = []
+        else:
+            skips = list(skips)
+
+        for action_class in resolve_action_order(self.actions):
+            # Decide if we need to skip because deps have failed
+            failed_deps = [d for d in action_class.dependencies if d not in successes]
+
+            action = action_class()
+
+            if failed_deps:
+                to_be = "was"
+                if len(failed_deps) > 1:
+                    to_be = "were"
+                message = "Skipped because %s %s not successful" % (format_sequence_as_message(failed_deps), to_be)
+
+                action.set_result(status="SKIP", error_id="SKIP", message=message)
+                skips.append(action)
+                continue
+
+            # Run the Action
+            try:
+                action.run()
+            except (Exception, SystemExit) as e:
+                # Uncaught exceptions are handled by constructing a generic
+                # failure message here that should be reported
+                message = (
+                    "Unhandled exception was caught: %s"
+                    "\nPlease file a bug at https://issues.redhat.com/ to have this"
+                    " fixed or a specific error message added."
+                    "\nTraceback: %s" % (e, traceback.format_exc())
+                )
+                action.set_result(status=STATUS_CODE["ERROR"], error_id="UNEXPECTED_ERROR", message=message)
+
+            # Categorize the results
+            if action.status <= STATUS_CODE["WARNING"]:
+                successes.append(action)
+
+            if action.status > STATUS_CODE["WARNING"]:
+                failures.append(action)
+
+        if self.next_stage:
+            successes, failures, skips = self.next_stage.run(successes, failures, skips)
+
+        return FinishedActions(successes, failures, skips)
+
+
+def resolve_action_order(potential_actions, previously_resolved_actions=None):
     """
     Order the Actions according to the order in which they need to run.
 
-    :arg potential_actions: Sequence of Actions which should be run.
-    :arg failed_actions: A set of failed Actions.  resolve_action_order() does not modify this but
-        does expect that its contents will be updated between iterations if there are more
-        failures.
-    :returns: A list of Actions sorted so that all dependent Actions are run before actions which
-        depend on their output.
-    :raises Exception: when it is impossible to satisfy a dependency in an Action.
+    :param potential_actions: Sequence of Actions which we need to find the
+        order of.
+    :type: Sequence
+    :param previously_resolved_actions: Sequence of Actions which have already
+        had been resolved into dependency order.
+    :returns: Iterator of Actions sorted so that all dependent Actions are run
+        before actions which depend on them.
+    :raises DependencyError: when it is impossible to satisfy a dependency in
```

Comment:
```suggestion
    :param potential_actions: Sequence of Actions which we need to find the
        order of.
    :type potential_actions: Sequence
    :param previously_resolved_actions: Sequence of Actions which have already
        had been resolved into dependency order.
    :type previously_resolved_actions: Sequence

    :raises DependencyError: when it is impossible to satisfy a dependency in
    
    :returns: Iterator of Actions sorted so that all dependent Actions are run
        before actions which depend on them.
```